src/uniforms/IESProfilesTexture.js

```diff
@@ -0,0 +1,127 @@
+import {
+	ClampToEdgeWrapping,
+	Color,
+	FloatType,
+	LinearFilter,
+	MeshBasicMaterial,
+	NoToneMapping,
+	RGBAFormat,
+	WebGLArrayRenderTarget,
+} from 'three';
+import { FullScreenQuad } from 'three/examples/jsm/postprocessing/Pass.js';
+import { IESLoader } from '../utils/IESLoader.js';
+
+const prevColor = new Color();
+export class IESProfilesTexture extends WebGLArrayRenderTarget {
+
+	constructor( ...args ) {
+
+		super( ...args );
+
+		const tex = this.texture;
+		tex.format = RGBAFormat;
+		tex.type = FloatType;
+		tex.minFilter = LinearFilter;
+		tex.magFilter = LinearFilter;
+		tex.wrapS = ClampToEdgeWrapping;
+		tex.wrapT = ClampToEdgeWrapping;
+		tex.generateMipmaps = false;
+
+		tex.updateFrom = ( ...args ) => {
+
+			this.updateFrom( ...args );
+
+		};
+
+		const fsQuad = new FullScreenQuad( new MeshBasicMaterial() );
+		this.fsQuad = fsQuad;
+
+		this.iesLoader = new IESLoader();
+
+	}
+
+	async updateFrom( renderer, iesProfiles ) {
+
+		const textures = [ ];
+		const promises = [ ];
+
+		iesProfiles.forEach( iesProfileURL => {
+
+			promises.push( new Promise( resolve => {
+
+				this.iesLoader
+					.load( iesProfileURL, texture => {
+
+						textures.push( texture );
+						resolve();
+
+					} );
+
+			} ) );
+
+		} );
```

Comment:
Similar to [RenderTarget2DArray](https://github.com/gkjohnson/three-gpu-pathtracer/blob/main/src/uniforms/RenderTarget2DArray.js) I think this class should keep the loading of IES profile textures separate from the initializing the texture itself. It might actually be cleanest for this class to extend `RenderTarget2DArray` since it looks like there's a lot of similar code.

Reply:
I think an issue there was that `RenderTarget2DArray` defaults to byte type and does some creation in the constructor, but I can always replace the constructor. That's still not very nice but is better than what I have here.

Reply:
It should be okay to override the defaults again for `format`, `type`, etc in constructor of `IESProfilesTexture`.